src/databricks/labs/ucx/source_code/notebook.py

```diff
@@ -225,6 +292,41 @@ def make_cell(lines_: list[str]):
 
         return cells
 
+    def _make_runnable(self, lines: list[str], cell_language: CellLanguage):
+        prefix = f"{self.comment_prefix} {MAGIC_PREFIX} "
+        prefix_len = len(prefix)
+        # pylint: disable=too-many-nested-blocks
+        for i, line in enumerate(lines):
+            if line.startswith(prefix):
+                line = line[prefix_len:]
+                if cell_language.requires_isolated_pi:
+                    if line.startswith(LANGUAGE_PREFIX):
+                        line = f"{cell_language.comment_prefix} {LANGUAGE_PI}"
+                lines[i] = line
+                continue
+            if line.startswith(self.comment_prefix):
+                line = f"{cell_language.comment_prefix} {COMMENT_PI}{line}"
+                lines[i] = line
+
+    def make_unrunnable(self, code: str, cell_language: CellLanguage) -> str:
```

Comment:
```suggestion
    def as_magic(self, code: str, cell_language: CellLanguage) -> str:
```
might be a better name

Reply:
done